File: SDKs/twilio.com/twilio_autopilot_v1/1.36.0/python/src/sdk/models/operations/listfield.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk.models import shared
LIST_FIELD_SERVERS = [
	"https://autopilot.twilio.com",
]


@dataclass
class ListFieldPathParams:
    assistant_sid: str = field(default=None, metadata={'path_param': { 'field_name': 'AssistantSid', 'style': 'simple', 'explode': False }})
    task_sid: str = field(default=None, metadata={'path_param': { 'field_name': 'TaskSid', 'style': 'simple', 'explode': False }})
    

@dataclass
class ListFieldQueryParams:
    page_size: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'PageSize', 'style': 'form', 'explode': True }})
    

@dataclass
class ListFieldSecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class ListFieldRequest:
    server_url: Optional[str] = field(default=None)
    path_params: ListFieldPathParams = field(default=None)
    query_params: ListFieldQueryParams = field(default=None)
    security: ListFieldSecurity = field(default=None)
    

@dataclass_json
@dataclass
class ListFieldListFieldResponseMeta:
    first_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'first_page_url' }})
    key: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'key' }})
    next_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'next_page_url' }})
    page: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page' }})
    page_size: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page_size' }})
    previous_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'previous_page_url' }})
    url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'url' }})
    

@dataclass_json
@dataclass
class ListFieldListFieldResponse:
    fields: Optional[List[shared.AutopilotV1AssistantTaskField]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'fields' }})
    meta: Optional[ListFieldListFieldResponseMeta] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'meta' }})
    

@dataclass
class ListFieldResponse:
    content_type: str = field(default=None)
    list_field_response: Optional[ListFieldListFieldResponse] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/twilio.com/twilio_autopilot_v1/1.36.0/typescript/src/sdk/models/operations/listfield.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const LISTFIELD_SERVERS = [
	"https://autopilot.twilio.com",
];



export class ListFieldPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=AssistantSid" })
  assistantSid: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=TaskSid" })
  taskSid: string;
}


export class ListFieldQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=PageSize" })
  pageSize?: number;
}


export class ListFieldSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class ListFieldRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  pathParams: ListFieldPathParams;

  @Metadata()
  queryParams: ListFieldQueryParams;

  @Metadata()
  security: ListFieldSecurity;
}


export class ListFieldListFieldResponseMeta extends SpeakeasyBase {
  @Metadata({ data: "json, name=first_page_url" })
  firstPageUrl?: string;

  @Metadata({ data: "json, name=key" })
  key?: string;

  @Metadata({ data: "json, name=next_page_url" })
  nextPageUrl?: string;

  @Metadata({ data: "json, name=page" })
  page?: number;

  @Metadata({ data: "json, name=page_size" })
  pageSize?: number;

  @Metadata({ data: "json, name=previous_page_url" })
  previousPageUrl?: string;

  @Metadata({ data: "json, name=url" })
  url?: string;
}


export class ListFieldListFieldResponse extends SpeakeasyBase {
  @Metadata({ data: "json, name=fields", elemType: shared.AutopilotV1AssistantTaskField })
  fields?: shared.AutopilotV1AssistantTaskField[];

  @Metadata({ data: "json, name=meta" })
  meta?: ListFieldListFieldResponseMeta;
}


export class ListFieldResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  listFieldResponse?: ListFieldListFieldResponse;

  @Metadata()
  statusCode: number;
}
